File: Retos/Reto #3 - EL GENERADOR DE CONTRASEÑAS [Media]/kotlin/roswer13.kt
fun main() {
    testCase(length = 6)
    testCase(length = 7)
    testCase(length = 8)
    testCase(length = 5)
    testCase(length = 9)
}

fun testCase(length: Int) {
    val password = generatePassword(length = length)

    if (password.length != length) {
        println(
            "Case with the length: '$length', returns ${password.length} but it should be $length"
        )
        return
    }

    println("Password: '$password', the length '${password.length}'")
}

fun generatePassword(length: Int): String {
    if (length in 6..8) {

        val chars = ('0'..'9') + ('a'..'z') + ('A'..'Z') + '*' + ',' + '.' + '$' + '&' + '/'

        return List(length) { chars.random() }.joinToString("")
    }

    return String()
}

File: Retos/Reto #3 - EL GENERADOR DE CONTRASEÑAS [Media]/dart/roswer13.dart
import 'dart:math';

void main() {
  testCase(length: 6);
  testCase(length: 7);
  testCase(length: 8);
  testCase(length: 5);
  testCase(length: 9);
}

void testCase({length = int}) {
  var password = generatePassword(length: length);

  if (password.length != length) {
    print(
        "Case with the length: '$length', returns ${password.length} but it should be $length");
    return;
  }

  print("Password: '$password', the length '${password.length}'");
}

String generatePassword({length = int}) {
  if (length >= 6 && length <= 8) {
    final letterLowerCase = "abcdefghijklmnopqrstuvwxyz";
    final letterUpperCase = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    final number = '0123456789';
    final special = '@#%^*>\$@?/[]=+';

    String chars = "";
    chars += '$letterLowerCase$letterUpperCase';
    chars += '$number';
    chars += '$special';

    return List.generate(length, (index) {
      final indexRandom = Random.secure().nextInt(chars.length);
      return chars[indexRandom];
    }).join('');
  }

  return '';
}
